pept/tracking/birmingham_method_gpu/birmingham_method_gpu.py

```diff
@@ -0,0 +1,427 @@
+        The algorithm will (over) estimate the amount of memory needed for the computation and
+        will allocate the arrays accordingly.  Memory will determine the amount of samples that
+        can be processed in parallel and therfore the amount of batches that will be processed.
+        The default value is 1.0, which means that the algorithm will use all the available memory.
+
+    threads_per_block : int
+        The number of threads per block. The default value is 1024. Depending on the compute
+        capability of the GPU, the maximum number of threads per block can be 256, 512 or 1024.
+
+
+
+    See Also
+    --------
+    pept.LineData : Encapsulate LoRs for ease of iteration and plotting.
+    pept.PointData : Encapsulate points for ease of iteration and plotting.
+    pept.utilities.read_csv : Fast CSV file reading into numpy arrays.
+    PlotlyGrapher : Easy, publication-ready plotting of PEPT-oriented data.
+    pept.scanners.ParallelScreens : Initialise a `pept.LineData` instance from
+                                    parallel screens PEPT detectors.
+
+    Examples
+    --------
+    A typical workflow would involve reading LoRs from a file, instantiating a
+    `BirminghamMethodGPU` class, tracking the tracer locations from the LoRs, and
+    plotting them.
+
+    >>> import pept
+    >>> from pept.tracking.BirminghamMethodGPU
+
+    >>> lors = pept.LineData(...)   # set sample_size and overlap appropriately
+    >>> bham = BirminghamMethodGPU()
+    >>> locations = bham.fit(lors)  # this is a `pept.PointData` instance
+
+    >>> grapher = PlotlyGrapher()
+    >>> grapher.add_points(locations)
+    >>> grapher.show()
+    '''
+
+    def __init__(self, fopt = 0.5, memory_usage = 1.0, threads_per_block = 1024):
+        '''`BirminghamMethodGPU` class constructor.
+
+        fopt : float, default 0.5
+            Float number between 0 and 1, representing the fraction of
+            remaining LORs in a sample used to locate the particle.
+
+        memory_usage : float
+            The percentage of GPU memory to be used by the algorithm. The value between 0 and 1.
+            The algorithm will (over) estimate the amount of memory needed for the computation and
+            will allocate the arrays accordingly.  Memory will determine the amount of samples that
+            can be processed in parallel and therfore the amount of batches that will be processed.
+            The default value is 1.0, which means that the algorithm will use all the available memory.
+
+        threads_per_block : int
+            The number of threads per block. The default value is 1024. Depending on the compute
+            capability of the GPU, the maximum number of threads per block can be 256, 512 or 1024.
+
+
+        '''
+        # First imort all necessery stuff
+        # gpu imports
+        # make imports global. There may be a better solution for this
+        global  cuda, SourceModule, DynamicSourceModule
+        try:

```

Comment:
I would add these imports at the top and let the code crash upon first use if the user doesn't have pycuda; you first use it when `cuda.Device.count()` - add the error message as a comment above that line